sources/sql_database/schema_types.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, Optional
+
+from sqlalchemy.sql import sqltypes
+from sqlalchemy import Table, Column
+
+from dlt.common.schema.typing import TColumnSchema, TTableSchemaColumns
+
+
+def sqla_col_to_column_schema(sql_col: Column[Any]) -> Optional[TColumnSchema]:
+    """Infer dlt schema column type from an sqlalchemy type.
+
+    Precision and scale is inferred from that types that support it,
+    such as numeric, varchar, int, bigint
+    """
+    sql_t = sql_col.type
+    col = None
+
+    if isinstance(sql_t, sqltypes.BigInteger):
+        col = dict(name=sql_col.name, data_type="bigint", precision=64)
+    elif isinstance(sql_t, sqltypes.SmallInteger):
+        col = dict(name=sql_col.name, data_type="bigint", precision=16)
+    elif isinstance(sql_t, sqltypes.Integer):
+        col = dict(name=sql_col.name, data_type="bigint", precision=32)
+    elif isinstance(sql_t, sqltypes.Numeric) and not isinstance(sql_t, sqltypes.Float):
+        col = dict(
+            name=sql_col.name,
+            data_type="decimal",
+            precision=sql_t.precision,
+            scale=sql_t.scale,
+        )
+    elif isinstance(sql_t, sqltypes.String):
+        col = dict(name=sql_col.name, data_type="text", precision=sql_t.length)
+    elif isinstance(sql_t, sqltypes._Binary):
+        col = dict(name=sql_col.name, data_type="binary", precision=sql_t.length)
+    elif isinstance(sql_t, sqltypes.DateTime):
```

Comment:
I'm not sure we should emit timestamp precision. it may just create incompatibilities and AFAIK sql alchemy uses datetime (floats) so precision is always microseconds anyway.
TLDR: I'd leave datetime and time precision

Reply:
True, makes sense. I've removed time/datetime